Assignments/A2.sql

```diff
@@ -0,0 +1,62 @@
+-- BDA 696 Assignment 2
+
+-- Use baseball database to create 3 tables
+
+use baseball;
+
+-- First table: Career Batting Average
+
+drop table if exists CareerBA;
+
+create table CareerBA
+select batter, sum(Hit)/sum(atBat) as BA
+from batter_counts where atBat>0
+group by batter;
+
+select * from CareerBA;
+
+-- Second table: Seasonal Batting Average
+
+drop table if exists SeasonalBA;
+
+create table SeasonalBA
+select batter_counts.batter, sum(batter_counts.Hit)/sum(batter_counts.atBat) as BA, 
+year(game.local_date) as Season
+from batter_counts join game on batter_counts.game_id=game.game_id where atBat>0
+group by Season, batter
+order by batter, Season;
+
+select * from SeasonalBA;
+
+-- Third table: Rolling Batting Average
+
+drop table if exists RollingBA;
+
+create table RollingBA
+select Hit, atBat, batter, date(game.local_date) as gameday,
+(sum(batter_counts.Hit) over (partition by batter
+						      order by game.local_date
+						      asc rows between 101 preceding
+						      and 1 preceding))
+						/
+(sum(batter_counts.atBat) over (partition by batter
+							    order by game.local_date
+							    asc rows between 101 preceding
+							    and 1 preceding))
+as rollingavg
```

Comment:
Maybe name this var `rollingavgBA` to include BA in the name for consistency and add career and season descriptive to the other BA vars to differentiate between them in tables. If you needed to join `CareerBA` and `SeasonalBA` later SQL would require you to rename the columns using `as` anyway, so could just do it here.

E.g.:

```
CREATE TABLE batter_join
SELECT CareerBA.batter,
	CareerBA.BA,
	SeasonalBA.BA
FROM CareerBA JOIN SeasonalBA ON CareerBA.batter = SeasonalBA.batter;
```
produces:
SQL Error [1060] [42S21]: (conn=70) Duplicate column name 'BA'

Reply:
Sounds good, renamed variables to cBA, sBA, and rBA to distinguish more clearly